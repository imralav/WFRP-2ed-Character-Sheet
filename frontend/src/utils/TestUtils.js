import React, {Fragment} from 'react';

import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';
import renderer from 'react-test-renderer';

import {BrowserRouter as Router, Route} from 'react-router-dom';

addLocaleData(en);

export default class TestUtils {
    static createComponentWithDefaultIntl(component) {
        return renderer.create(
            <IntlProvider locale="en" messages={{}}>
                {component}
            </IntlProvider>
        );
    }

    static createComponentWithDefaultIntlAndRouter(component) {
        return TestUtils.createComponentWithDefaultIntl(
            <Router>
                <Fragment>
                {component};
                </Fragment>
            </Router>
        );
    }
}