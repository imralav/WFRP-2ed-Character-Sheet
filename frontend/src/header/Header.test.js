import React from 'react';
import Header from '../header';
import TestUtils from '../utils/TestUtils';

describe('Header', () => {
    let component;
    let componentRoot;
    let tree;
    let changeLocaleFunction;
    beforeEach(() => {
        changeLocaleFunction = jest.fn();
        component = TestUtils.createComponentWithDefaultIntlAndRouter(
            <Header changeLocale={locale => changeLocaleFunction(locale)}/>
        );
        componentRoot = component.root;
        tree = component.toJSON();
    });
    it('should match previous snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
    describe('changing locale', () => {
        [{label: 'EN', locale: 'en'}, {label: 'PL', locale: 'pl'}].forEach(localeData => {
            describe(`for ${localeData.label}`, () => {
                let button;
                beforeEach(() => {
                    button = componentRoot.findByProps({children: localeData.label});
                    button.props.onClick();
                });
                it(`should change locale to ${localeData.locale}`, () => {
                    expect(changeLocaleFunction).toHaveBeenCalledWith(localeData.locale);
                });
            });
        });
    });
});
