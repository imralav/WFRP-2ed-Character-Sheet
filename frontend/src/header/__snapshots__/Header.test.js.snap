// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match previous snapshot 1`] = `
Array [
  <div
    className="row header"
>
    <div
        className="col-3"
    >
        <a
            href="/"
            onClick={[Function]}
        >
            <span>
                Home
            </span>
        </a>
    </div>
    <div
        className="col-9 text-right"
    >
        <a
            className="localeChangeButton"
            onClick={[Function]}
        >
            EN
        </a>
        <a
            className="localeChangeButton"
            onClick={[Function]}
        >
            PL
        </a>
    </div>
</div>,
  ";",
]
`;
